website/docs/plugin/framework/migrating/resources/index.mdx

```diff
@@ -0,0 +1,222 @@
+- SDKv2 implements a resource's CRUD operations as functions on the `schema.Resource`. In the Framework, you define a
+type that implements the `Resource` interface. The resource interface contains the functions that define your resource's
+CRUD operations.
+
+## Example
+
+The following examples show how to migrate portions of the
+[tls](https://github.com/hashicorp/terraform-provider-tls) provider.
+
+For a complete example,
+clone the `terraform-provider-tls` repository and compare the `resource_private_key.go` file in
+[v3.4.0](https://github.com/hashicorp/terraform-provider-tls/blob/v3.4.0/internal/provider/resource_private_key.go) with
+[v4.0.1](https://github.com/hashicorp/terraform-provider-tls/blob/v4.0.1/internal/provider/resource_private_key.go).
+
+### SDKv2
+
+In SDKv2, the `ResourcesMap` field on the `schema.Provider` struct holds a `map[string]*schemaResource`. A typical
+pattern is to implement a function that returns `schema.Resource`.
+
+The following example from the `provider.go` file defines a `tls_private_key` resource within the provider schema.
+
+```go
+func New() (*schema.Provider, error) {
+    return &schema.Provider {
+        ResourcesMap: map[string]*schema.Resource {
+            "tls_private_key": resourcePrivateKey(),
+            /* ... */
+```
+
+The following example from the `resource_private_key.go` file defines the resource schema.
+
+```go
+func resourcePrivateKey() *schema.Resource {
+    return &schema.Resource{
+        CreateContext: createResourcePrivateKey,
+        DeleteContext: deleteResourcePrivateKey,
+        ReadContext:   readResourcePrivateKey,
+
+        Description: "Creates a PEM /* ... */",
+
+        Schema: map[string]*schema.Schema{
+            "algorithm": {
+                Type:             schema.TypeString,
+                Required:         true,
+                ForceNew:         true,
+                ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(SupportedAlgorithmsStr(), false)),
+                Description: "Name of the algorithm to use when generating the private key. " +
+                    "Currently-supported values are `RSA`, `ECDSA` and `ED25519`.",
+            },
+            /* ... */
+```
+
+### Framework
+
+The following shows the same section of provider code after the migration.
+
+This code defines the `tls_private_key` resource by mapping the resource name to the `privateKeyResourceType` struct.
+
+```go
+func (p *provider) GetResources(_ context.Context) (map[string]provider.ResourceType, diag.Diagnostics) {
+    return map[string]provider.ResourceType{
+        "tls_private_key":         &privateKeyResourceType{},
+        /* ... */
+```
+
+This code defines the `GetSchema` and `NewResource` functions for the `privateKeyResourceType`.
+
+```go
+func (rt *privateKeyResourceType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+    return tfsdk.Schema{
+        Version: 1,
```

Comment:
I'm guessing that the sdk/v2 code also included `Version: 1` -- we may want to show it above if that's the case (or conversely remove this here) so folks don't think they _must_ use `Version: 1` or higher in the framework 👍

Reply:
Have removed.